cpp/test/core/blob_tests.cpp

```diff
@@ -33,6 +33,79 @@ TEST_CASE_BEGIN(LargeBlob)
 }
 TEST_CASE_END
 
+void check_blob_is_empty(const bond::blob& b)
+{
+    // workaround: old compilers won't let us print nullptr, while new ones won't
+    // compare nullptrs to ints
+
+    void* null = 0;
+    UT_AssertAreEqual(b.data(), null);
+    UT_AssertAreEqual(b.size(), 0u);
+    UT_AssertAreEqual(b.content(), null);
+}
+
+TEST_CASE_BEGIN(BlobMoveAssign)
+{
+    using bond::blob;
+
+    const uint32_t size = 32;
+
+    // move default-constructed blob
+    {
+        auto data = boost::make_shared<std::array<char,size>>();
+        blob a{data, size};
+        blob b;
+
+        UT_AssertAreEqual(data.use_count(), 2);
+
+        a = std::move(b);
+        
+        check_blob_is_empty(a);
+        check_blob_is_empty(b);
+
+        UT_AssertAreEqual(data.use_count(), 1);
+    }
+    // move from instance with shared ptr
+    {
+        // move from other instance
+        auto data = boost::make_shared<std::array<char, size>>();
+        blob a;
+        blob b{data, size};
+
+        UT_AssertAreEqual(data.use_count(), 2);
+
+        a = std::move(b);
+
+        UT_AssertAreEqual(a.data(), data.get());
+        UT_AssertAreEqual(a.size(), size);
+
+        check_blob_is_empty(b);
+
+        UT_AssertAreEqual(data.use_count(), 2);
+    }
+    // move from raw memory
+    {
+        std::unique_ptr<std::array<char, size>> p1(new std::array<char, size>);
```

Comment:
Why do you need `unique_ptr` here? Can't you just keep them on the stack (they are small enough)?